BinaryEncoding.md

```diff
@@ -1,122 +1,408 @@
+| f32 count | `uint16` | `flags[2] == 1` | number of `f32` local variables |
+| f64 count | `uint16` | `flags[2] == 1` | number of `f64` local variables |
+| body size | `uint16` | `flags[0] == 0` | size of function body to follow, in bytes |
+| body | `bytes` | `flags[0] == 0` | function body |
+
+### Data Segments section
+The data segemnts section declares the initialized data that should be loaded into the linear memory.
+A module may only contain one data segments section.
+
+| Field | Type | Description |
+| ----- |  ----- | ----- | 
+| id = `0x04` | `uint8` | section identifier for data segments |
+| count | `varuint32` | count of data segments to follow | 
+| entries | `data_segment*` | repeated data segments as described below |
+
+* ```varuint32```: The number of data segments in the section.
+* For each data segment:
+  - ```uint32```: The base address of the data segment in memory.
+  - ```uint32```: The offset of the data segment's data in the file.
+  - ```uint32```: The size of the data segment (in bytes)
+  - ```uint8```: ```1``` if the segment's data should be automatically loaded into memory at module load time.
+
+### Indirect Function Table section
+The indirect function table section declares the size and entries of the indirect function table, which consist
+of indexes into the [Functions](#functions-section) section.
+This section must be preceded by a [Functions](#functions-section) section.
+
+| Field | Type | Description |
+| ----- |  ----- | ----- | 
+| id = `0x05` | `uint8` | section identifier for indirect function table |
+| count | `varuint32` | count of entries to follow | 
+| entries | `uint16*` | repeated indexes into the function table |
+
+### End section
+This indicates the end of the sections. Additional data that follows this section marker is not interpreted
+by the decoder. It can used, for example, to store function names or data segment bodies.
+
+| Field | Type | Description |
+| ----- |  ----- | ----- | 
+| id = `0x06` | `uint8` | section identifier for end of sections |
+
+
+### Nonstandard: Globals section
+A module may only contain one globals section. This section is currently for V8 internal use.
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| id = `0x03` | `uint8` | section identifier for globals |
+| count | `varuint32` | count of global variable entries to follow | 
+| entries | `global_variable*` | repeated global variable entries as described below |
+
+#### Global Variable entry
+
+A global variable entry describes a variable outside the WASM linear memory. A Global variable can only be loaded and
+stored to from dedicated instructions.
+
+| Field | Type | Description |
+| ----- |  ----- | ----- | 
+| name | `uint32` | name of the global variable as an offset to a string in the module | 
+| type | `uint8` | the type of the global, as a memory type |
+| exported | `uint8` | a boolean indicating whether the global variable is exported |
+
+### WorkInProgress: WLL section
```

Comment:
@JSStats 

Let's keep this discourse polite and treat each other with respect. CC'ing or threatening to CC the CEO of a corporation over a minor disagreement w.r.t. a binary format is neither going to be productive (our organizations do not work like that) nor is it acceptable behavior (it is aggressive and threatening). It saddens me to observe that despite your recent and welcome cooperativeness, untoward escalation and threats are appearing again. I personally accepted your patch to V8 to allow the WLL section because it was "no big deal" and included it on this PR along the vein of documenting the existing format. An open and dispassionate discussion about whether to include it is entirely expected and welcome. Threats are not.